precompiles/bank/method_withdraw.go

```diff
@@ -0,0 +1,198 @@
+package bank
+
+import (
+	"math/big"
+
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+
+	ptypes "github.com/zeta-chain/node/precompiles/types"
+	"github.com/zeta-chain/node/x/fungible/types"
+)
+
+// withdraw is used to withdraw cosmos coins minted using the bank's deposit function.
+// The caller has to have enough cosmos coin on its cosmos account balance to withdraw the requested amount.
+// After all check pass the bank will burn the cosmos coins and transfer the ZRC20 amount to the withdrawer.
+// The cosmos coins have the denomination of "zrc20/0x12345" where 0x12345 is the ZRC20 address.
+// Call this function using solidity with the following signature:
+// From IBank.sol: function withdraw(address zrc20, uint256 amount) external returns (bool success);
+// The address to be passed to the function is the ZRC20 address, like in 0x12345.
+func (c *Contract) withdraw(
+	ctx sdk.Context,
+	evm *vm.EVM,
+	contract *vm.Contract,
+	method *abi.Method,
+	args []interface{},
+) (result []byte, err error) {
+	// 1. Check everything is correct.
+	if len(args) != 2 {
+		return nil, &(ptypes.ErrInvalidNumberOfArgs{
+			Got:    len(args),
+			Expect: 2,
+		})
+	}
+
+	// Unpack parameters for function withdraw.
+	// function withdraw(address zrc20, uint256 amount) external returns (bool success);
+	zrc20Addr, amount, err := unpackWithdrawArgs(args)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the correct caller address.
+	caller, err := getEVMCallerAddress(evm, contract)
```

Comment:
Should we disable delegate calls in the precompile to make things simpler? Not sure how to check if it's a delegate call, maybe when the address in context doesn't match `contract.Address()`?

Reply:
This would not allow upgradable contract to call preccompile then?